### PR TITLE
Added middleman 4.0 compatibility

### LINKED ISSUE #6: middleman 4.0 support
hello,

does this exentsion support middleman 4.0. I got the following error message:

```
middleman-search was resolved to 0.0.1, which depends on
  middleman-core (~> 3.2)
```

and the middleman server crashed with the following error message:

```
== The Middleman is loading
/home/xx/.gem/ruby/2.2.0/gems/middleman-core-4.0.0/lib/middleman-core/extensions.rb:86:in `load': Unknown Extension: search. Check the name and make sure you have referenced the extension's gem in your Gemfile. (RuntimeError)
```

### PR DESCRIPTION
**Pull request by drallgood**
This PR allows this plugin to be used with middleman 4.0 (fixes #6)

### CODE REVIEW

File: middleman-search.gemspec [hunk 1]
```diff
@@ -18,9 +18,9 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "middleman-core", ["~> 3.2"]
-  spec.add_dependency "therubyracer", ["~> 0.12"]
-  spec.add_dependency "nokogiri", ["~> 1.5"]
+  spec.add_dependency "middleman-core", ["~> 4.0"]
```
Comment (spalladino) [inline middleman-search.gemspec]: Thanks for the PR! We'd like to keep support for 3.x, and this change would force users to work with 4.0. Do you think you could do something like https://github.com/middleman/middleman-livereload/commit/4f619cf5aa0f8fc2f307196f79a4ffeefd04bae9#diff-4d67daa0283bf974174fb193b9560c18 to accept both majors?

Comment (drallgood) [inline middleman-search.gemspec]: @spalladino Sure.

**Comment by drallgood** [comment]
@annael Any news on this issue?

**Comment by matiasgarciaisaia** [comment]
:+1: Works for me. I'll be releasing a new version of the gem with this.

Thank you!

**Comment by drallgood** [comment]
Glad I could help :smile:

**Pull request merged**